include/boost/astronomy/coordinate/cartesian_differential.hpp

```diff
@@ -1,141 +1,233 @@
 #ifndef BOOST_ASTRONOMY_COORDINATE_CARTESIAN_DIFFERENTIAL_HPP
 #define BOOST_ASTRONOMY_COORDINATE_CARTESIAN_DIFFERENTIAL_HPP
 
+#include <tuple>
+#include <type_traits>
+
+#include <boost/geometry/strategies/strategy_transform.hpp>
 #include <boost/geometry/geometries/point.hpp>
 #include <boost/geometry/algorithms/transform.hpp>
 #include <boost/geometry/core/cs.hpp>
 #include <boost/static_assert.hpp>
+#include <boost/units/get_dimension.hpp>
+#include <boost/units/systems/si/length.hpp>
+#include <boost/units/systems/si/time.hpp>
+#include <boost/units/systems/si/velocity.hpp>
 
 #include <boost/astronomy/detail/is_base_template_of.hpp>
 #include <boost/astronomy/coordinate/base_differential.hpp>
 
 
-namespace boost
+namespace boost { namespace astronomy { namespace coordinate {
+
+namespace bu = boost::units;
+
+//!Represents differential of cartesian representation
+//!Uses three coordinate to represent a differential (dx, dy, dz)
+template
+<
+    typename CoordinateType = double,
+    typename XQuantity = bu::quantity<bu::si::velocity, CoordinateType>,
+    typename YQuantity = bu::quantity<bu::si::velocity, CoordinateType>,
+    typename ZQuantity = bu::quantity<bu::si::velocity, CoordinateType>
+>
+struct cartesian_differential: base_differential<3, bg::cs::cartesian, CoordinateType>
 {
-    namespace astronomy
+    ///@cond INTERNAL
+    BOOST_STATIC_ASSERT_MSG((std::is_same<typename bu::get_dimension<XQuantity>::type,
```

Comment:
instead of checking the dimensions for velocity check if all component has the same dimension.